docs/sources/flow/reference/components/prometheus.exporter.azure.md

```diff
@@ -77,6 +90,8 @@ Tags in `included_resource_tags` will be added as labels with the name `tag_<tag
 
 Valid values for `azure_cloud_environment` are `azurecloud`, `azurechinacloud`, `azuregovernmentcloud` and `azurepprivatecloud`.
 
+`validate_dimensions` is disabled by default to reduce the number of azure exporter instances requires when a `resource_type` has metrics with varying dimensions. When `validate_dimensions` is enabled you will need 1 exporter instance per metric + dimension combination which is more tedious to maintain.  
+
```

Comment:
is this a breaking change?

Reply:
Maybe? This used to default to `true` so any existing configs won't break but the default behavior will change. I could flip the default to `true` to match the existing behavior or add it in the breaking changes section.

Reply:
considering the feature, I think it's better to leave the false default and do a breaking change. wdyt @tpaschalis @mattdurham ?

Reply:
From what I understand, it would only mean existing configs could potentially start accepting more dimensions although they would be manually specified anyway

Reply:
Whats the scope of the change between the two? Feels like we are getting more metrics, are we talking 2x,10x,100x? Consider me very ignorant of the internal workings of azure :)

Reply:
For some additional context, if it makes sense to do a breaking change then I am onboard with it but it would need to be documented.

Reply:
@mattdurham I don't know much either about azure internals but I'm thinking the "break" is that wrong configs would start working. Say if someone has something like this and the agent is running, it would start working catching the default dimensions for each metric.
```
prometheus.exporter.azure "example" {
	subscriptions = ["<...>"]
	resource_type = "Microsoft.Network/applicationGateways"

	included_dimensions = []  
	metrics             = [
		"BackendResponseStatus",
		"Throughput",
	]
	timespan = "PT1M"
}
```

Reply:
This might be more of a behavioral change vs a breaking change. It won't change any currently working configuration someone will be running. They will still get the same amount of data. 

On main a config which requests this data from azure fails but runs without issue on this branch,
```yaml
            resource_type: ["microsoft.containerservice/managedclusters"]
            metrics:
              - node_cpu_usage_millicores
              - node_cpu_usage_percentage
              - node_disk_usage_bytes
              - node_disk_usage_percentage
              - node_memory_rss_bytes
              - node_memory_rss_percentage
              - node_memory_working_set_bytes
              - node_memory_working_set_percentage
              - node_network_in_bytes
              - node_network_out_bytes
            included_resource_tags:
              - environment
            included_dimensions:
              - node
              - nodepool
              - device
```
The `device` dimension is only valid for some of the requested metrics, https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-containerservice-managedclusters-metrics. The only valid way to do this on main was to split the config in to multiple definitions (do a self scrape in static mode or multiple exporter definitions in flow mode). 

Without validation though there's no safe guards for mistyping/misconfiguration so accidentally doing something like `devie` would not give you an error where it would previously.

Reply:
Sounds reasonable then and pretty fine with this then